### PR TITLE
Build Schema & Schema Printer

### PR DESCRIPTION
**Pull request by petecoop**
I've made some progress on #90 and thought I'd open a PR so tests run etc.

So far I've only done 2 tests and need to do Docblocks, it looks like I also need to write a Schema Printer too to match the js tests.

### CODE REVIEW

**Comment by vladar** [comment]
Looks ok to me, except for tests as coverage has dropped significantly.

**Comment by petecoop** [comment]
Hey @vladar yeah not finished just yet, tests are not done for `BuildSchema` as I had to write `SchemaPrinter` first, working on it!

**Comment by petecoop** [comment]
Hey @vladar, this is now finished! All the tests are identical to `graphql-js`. And everything follows `graphql-js` almost 1:1.

I did have some trouble with something I found in the lexer - where you're using `substr()` put it's not technically identical to the js equivalent of `slice()`, I tried to fix it in the lexer but it caused other breaking tests elsewhere, so worked around it instead.

File: src/Utils/SchemaPrinter.php [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    }
+
+    private static function isIntrospectionType($typename)
+    {
+        return strpos($typename, '__') === 0;
+    }
+
+    private static function isBuiltInScalar($typename)
+    {
+        return (
```
Comment (mcg-web) [inline src/Utils/SchemaPrinter.php]: Use [constants](https://github.com/webonyx/graphql-php/blob/master/src/Type/Definition/Type.php#L20-L24) please :)

Comment (petecoop) [inline src/Utils/SchemaPrinter.php]: done!

**Comment by vladar** [comment]
Thanks for your work! I will publish 0.9.4 release soon with this feature included.

File: src/Executor/Values.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ public static function getVariableValues(Schema $schema, $definitionNodes, array
      * @return array
      * @throws Error
      */
-    public static function getArgumentValues($def, $node, $variableValues)
+    public static function getArgumentValues($def, $node, $variableValues = null)
```
Comment (mcg-web) [inline src/Executor/Values.php]: @vladar here is a BC this should not be released in a minor version?

Comment (vladar) [inline src/Executor/Values.php]: It is a relaxed requirement, all existing calls to this method will work as they do now. I don't see how it is a BC break?

Comment (mcg-web) [inline src/Executor/Values.php]: its a BC because the signature change, this class is not final then it can be override :)

Comment (vladar) [inline src/Executor/Values.php]: In theory this is possible, but we use this class as a container for pure stateless functions and reference these static methods directly by class name, so it doesn't imply any polimorphism. It is implicitly final. But yeah - we should probably mark such classes as final explicitly.

Anyway it's too late: I released 0.9.4 even before your first comment %) But I am sure it won't break anyone, so let's cross the fingers %)

**Pull request merged**